### ISSUE TITLE
window movement

### ISSUE THREAD

**Issue by rrahl0** [pop-os/cosmic-comp#528]
When I move a maximized window from one virtual desktop to another, it should keep it's maximized status. As of right now, it gets unmaximized (but it stays the same size (gets the active hint borders)

**Comment by RafaelKr** [comment]
This also happens when moving windows between displays (via Shortcut `Shift + SUPER + Alt + Direction keys`) or Workspaces (via Shortcut `Shift + SUPER + Ctrl + Direction keys` or `Shift + SUPER + 0-9 keys`)

**Comment by RafaelKr** [comment]
@ids1024 you fixed #1645 and #1819 via https://github.com/pop-os/cosmic-comp/commit/612ff2f523580af2729cc7de1e73c334ea135726

This issue is somehow related, it would be nice if you could have a look at this also. Would be a great usability improvement, as I'm often moving maximized windows between my screens.

**Comment by leviport** [comment]
I don't think those are related to this in any meaningful way.

**Comment by RafaelKr** [comment]
From my perspective as a user, it seemed as though there might be some overlap in the underlying code paths, since all of the issues relate to the size/status of windows after moving them.

Therefore, I thought it would be useful to have someone who is currently familiar with this part of the code take a look at it.

I apologise if my comments are distracting.

**Comment by notlaast** [comment]
TLDR: two behaviors should be fixed:
1. I agree that a keyboard shortcut which pushes a maximized window to a new monitor should retain its maximized state; but,
2. dragging a maximized window should indeed produce a non-maximized window with some particular size, and that size should *not* change after drag-and-drop is ended.

I don't mean to change scope but the title of this ticket is vague and I think clarifications are worth noting here. I agree with others above about only one aspect of the general behavior, and just hope the other undesirable behavior is also addressed.

To describe the second at the risk of being pedantic: I think dragging should indeed retain the window size but not the maximized state (and it's anyhow relatively easy to drag it to the top of the new screen to become maximized again). This behavior is, however, not perfect: the dragged window initially takes on its previous non-maximized size while being dragged, but then it gets auto-resized when I stop dragging so that it then jumps to its previous size while it was still maximized (except now no longer in the maximized state). Case in point: when I drag a maximized window from my smaller laptop screen directly to the bottom-right corner of larger screen, the window inevitably snaps to some other size which is no longer flush with the screen edges.
